ocaml/xapi/message_forwarding.ml

```diff
@@ -629,6 +629,14 @@ functor
           Ref.string_of repository
       with _ -> "invalid"
 
+    let rpm_gpg_key_uuid ~__context rpm_gpg_key =
+      try
+        if Pool_role.is_master () then
+          Db.Rpm_gpg_key.get_uuid ~__context ~self:rpm_gpg_key
+        else
+          Ref.string_of rpm_gpg_key
+      with _ -> "invalid"
```

Comment:
Is this a good way to handle this? Should this not raise an exception rather than returning some malformed string?